src/Ads/AssetSuggestionsService.php

```diff
@@ -0,0 +1,211 @@
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Proxies\WP;
+
+/**
+ * Class AssetSuggestionsService
+ *
+ * Suggest assets and possible final URLs.
+ *
+ * @since x.x.x
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Ads
+ */
+class AssetSuggestionsService implements Service {
+
+	/**
+	 * AssetSuggestionsService constructor.
+	 *
+	 * @param WP $wp
+	 */
+	public function __construct( WP $wp ) {
+		$this->wp = $wp;
+	}
+
+	/**
+	 * Get posts that can be used to suggest assets
+	 *
+	 * @param string $search The search query.
+	 * @param int    $per_page Number of items per page.
+	 * @param int    $offset Used in the get_posts query.
+	 *
+	 * @return array
+	 */
+	protected function get_post_suggestions( $search, $per_page, $offset = 0 ): array {
+		$post_suggestions    = [];
+		$excluded_post_types = [ 'attachment' ];
+
+		$post_types = $this->wp->get_post_types(
+			[
+				'exclude_from_search' => false,
+				'public'              => true,
+			]
+		);
+
+		// Exclude attachment post_type
+		$filtered_post_types = array_diff( $post_types, $excluded_post_types );
+
+		$args = [
+			'post_type'      => $filtered_post_types,
+			'posts_per_page' => $per_page,
+			'post_status'    => 'publish',
+			's'              => $search,
+			'offset'         => $offset,
+		];
+
+		$posts = $this->wp->get_posts( $args );
+
+		foreach ( $posts as $post ) {
+			$post_suggestions[] = $this->format_final_url_response( $post->ID, 'post', $post->post_title, get_permalink( $post->ID ) );
+		}
+
+		return $post_suggestions;
+	}
+
+	/**
+	 * Get terms that can be used to suggest assets
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 *
+	 * @return array
+	 */
+	protected function get_terms_suggestion( $search, $per_page ): array {
+		$terms_suggestions = [];
+
+		// Get all taxonomies that are public, show_in_menu = true helps to exclude taxonomies such as "product_shipping_class".
+		$taxonomies = $this->wp->get_taxonomies(
+			[
+				'public'       => true,
+				'show_in_menu' => true,
+			],
+		);
+
+		$terms = $this->wp->get_terms(
+			[
+				'taxonomy'   => $taxonomies,
+				'hide_empty' => false,
+				'number'     => $per_page,
+				'name__like' => $search,
+			]
+		);
+
+		foreach ( $terms as $term ) {
+				$terms_suggestions[] = $this->format_final_url_response( $term->term_id, 'term', $term->name, get_term_link( $term->term_id, $term->taxonomy ) );
+		}
+
+		return $terms_suggestions;
+
+	}
+
+
+	/**
+	 * Return a list of final urls that can be used to suggest assets.
+	 *
+	 * @param string $search The search query
+	 * @param int    $per_page Number of items per page
+	 * @param string $order_by Order by: id, type, title, url
+	 *
+	 * @return array final urls with their title, id & type.
+	 */
+	public function get_final_urls_suggestions( $search = '', $per_page = 30, $order_by = 'title' ): array {
```

Comment:
💅  minor

Same as 
https://github.com/woocommerce/google-listings-and-ads/pull/1784/files#r1031025484